derive_client/_bridge/transaction.py

```diff
@@ -4,28 +4,20 @@
 from web3 import Web3
 from web3.contract import Contract
 
-from derive_client.constants import DEFAULT_GAS_FUNDING_AMOUNT, DEPOSIT_GAS_LIMIT, MSG_GAS_LIMIT
-from derive_client.data_types import Address, ChainID, TxStatus
-from derive_client.exceptions import InsufficientGas
-from derive_client.utils import build_standard_transaction, estimate_fees, exp_backoff_retry, send_and_confirm_tx
+from derive_client.data_types import Address, TxStatus
+from derive_client.exceptions import InsufficientTokenBalance
+from derive_client.utils import build_standard_transaction, send_and_confirm_tx
 
 
-def _check_gas_balance(w3: Web3, account: Address, gas_limit=DEPOSIT_GAS_LIMIT):
-    """Check whether the account has sufficient gas balance."""
-    balance = w3.eth.get_balance(account)
-    if balance < gas_limit:
-        raise InsufficientGas(
-            f"Insufficient balance for gas: {gas_limit} < {balance} ({(balance / gas_limit * 100):.2f}%)"
-        )
```

Comment:
this was just shit: we don't know the gas cost in advance, and `DEPOSIT_GAS_LIMIT` is an incorrect proxy. Instead, we check in the `simulate_tx` (already) and raise `InsufficientNativeBalance` or fail the simulation after (or the tx submission itself after potentially)